test/test_cfstring.cpp

```diff
@@ -0,0 +1,37 @@
+/*  Copyright (c) 2016 Michael Hansen
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE. */
+
+#include "st_string_darwin.h"
+
+#include <gtest/gtest.h>
+
+TEST(string, CFString_compatibility)
+{
+    CFStringRef cfStr = CFSTR("Hello World");
+    ST::string stStr(cfStr);
+    EXPECT_EQ(stStr, ST_LITERAL("Hello World"));
+
+    ST::string s = ST::string::from_CFString(CFSTR("Testing"));
+    ST::string s2 = ST_LITERAL("Testing");
+    CFStringRef s_cf = s2.to_CFString();
+    //EXPECT_EQ(1, CFGetRetainCount(s_cf));
```

Comment:
On newer versions of macOS, `CFGetRetainCount` seems to always return -1 regardless of the actual ref count value, which makes testing the ownership annotations for accuracy super hard 😒

Reply:
Testing via retain count is extremely recommended against by Apple because their own private code may affect the retain count. (For example, they may have internal autorelease pools retaining and releasing the ref on their own schedule.) Retain count should never be used for unit tests because the result can vary by OS version depending on the internal implementation.

It's possible they got tired of people trying to monitor retain counts and yanked the function.